### PR TITLE
[Tour Of Beam] API adjustments

### PR DESCRIPTION
**Pull request by eantyshev**
A collection of minor changes in API
addresses #23400

 - [x] Decouple SDK Id and Title, return Sdk objects in getSdkList response 
 - [x] Return `title` instead of `name` in getContentTree response
 - [x] Return `id` instead of `unitId/moduleId` in getContentTree response   
 - [x] Set CORS: * response header 

Per-commit review is strongly suggested

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by vchunikhin** [comment]
lgfm

@eantyshev hi!, look through that https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Commit/8063/console
PreCommit was failed.

**Comment by eantyshev** [comment]
That was a glitch in a CI infrastructure

File: learning/tour-of-beam/backend/docker-compose.yml [hunk 1]
```diff
@@ -25,4 +25,4 @@ services:
       - DATASTORE_LISTEN_ADDRESS=0.0.0.0:8081
     ports:
       - "8081:8081"
-    command: --consistency=1.0
+    command: --consistency=1.0 --store-on-disk
```
Comment (olehborysevych) [inline learning/tour-of-beam/backend/docker-compose.yml:28]: @eantyshev maybe we should try to set consistency to recommended 0.05 for development?

Comment (olehborysevych) [reply learning/tour-of-beam/backend/docker-compose.yml:28]: @eantyshev  --store-on-disk is a enabled by default not sure we need it here

Comment (eantyshev) [reply learning/tour-of-beam/backend/docker-compose.yml:28]: > --store-on-disk is a enabled by default not sure we need it here
I believe better to set it explicitly, provided we have to define other cmd options, as the default may change in future

> maybe we should try to set consistency to recommended 0.05 for development?
This setup is about CI/CD, we don't want to introduce undefined behavior here

**Comment by olehborysevych** [comment]
R: @damccorm

File: learning/tour-of-beam/backend/middleware.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+// middleware to parse sdk query param and pass it as additional handler param.
+func ParseSdkParam(next HandlerFuncWithSdk) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sdkStr := r.URL.Query().Get("sdk")
+		sdk := tob.ParseSdk(sdkStr)
+
+		if sdk == tob.SDK_UNDEFINED {
+			log.Printf("Bad sdk: %v", sdkStr)
+			finalizeErrResponse(w, http.StatusBadRequest, BAD_FORMAT, "unknown sdk")
```
Comment (damccorm) [inline learning/tour-of-beam/backend/middleware.go:73]: Any reason we took out the message about available SDKs here? (`message := fmt.Sprintf("Sdk not in: %v", tob.SdksList())`)

File: learning/tour-of-beam/backend/internal/sdk.go [hunk 3]
```diff
@@ -19,33 +19,52 @@ type Sdk string
...
 
 func (s Sdk) String() string {
 	return string(s)
 }
 
-// Parse sdk from string names, f.e. "Java" -> Sdk.GO_JAVA
+func (s Sdk) Title() string {
+	switch s {
+	case SDK_GO:
+		return "Go"
```
Comment (damccorm) [inline learning/tour-of-beam/backend/internal/sdk.go:35]: Any reason we capitalize names here and leave them all lowercase below?

File: learning/tour-of-beam/backend/internal/sdk.go [hunk 4]
```diff
@@ -19,33 +19,52 @@ type Sdk string
...
+	SDK_JAVA      Sdk = "java"
+	SDK_SCIO      Sdk = "scio"
 )
 
 func (s Sdk) String() string {
 	return string(s)
 }
 
-// Parse sdk from string names, f.e. "Java" -> Sdk.GO_JAVA
+func (s Sdk) Title() string {
```
Comment (damccorm) [inline learning/tour-of-beam/backend/internal/sdk.go:32]: Could you add a function header comment?

**Review by damccorm — COMMENTED** [review]
Overall LGTM, just had a few questions/comments

File: learning/tour-of-beam/backend/middleware.go (hunk 2 above)
Comment (eantyshev) [reply learning/tour-of-beam/backend/middleware.go:73]: Just realized that it makes no sense responding with SDK value to the client: it has just given it as a query parameter

File: learning/tour-of-beam/backend/internal/sdk.go (hunk 3 above)
Comment (damccorm) [reply learning/tour-of-beam/backend/internal/sdk.go:36]: SGTM

**Comment by damccorm** [comment]
Looks like we need to run go fmt for precommits to pass

**Review by damccorm — APPROVED** [review]
LGTM - thanks!

**Pull request merged**